cdemu: migrate to by-name and refactor definitions

guylamar2006 opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

bendlas reviewed (commented):
TBH I'm not sure how I feel about scattering these packages into different folders. I don't think anything re-uses any of them and they are usually released together. So if nothing else, splitting these up will make checking the update status slightly more incovenient.

But all of that might just be me, being sentimental, because cdemu was my first contribution to nixos. So I'll agree and approve.

I'll not review all of it immediately, because I'm currently mobile. Splitting modifications and moves into separate commits would make this somewhat more easy.

guylamar2006 committed:
image-analyzer: migrate to by-name

guylamar2006 committed:
libmirage: migrate to by-name

guylamar2006 committed:
gcdemu: migrate to by-name

guylamar2006 committed:
cdemu-daemon: migrate to by-name

guylamar2006 committed:
cdemu-client: migrate to by-name

guylamar2006 committed:
cdemu-daemon: refactor

guylamar2006 committed:
cdemu-client: refactor

guylamar2006 committed:
image-analyzer: refactor

guylamar2006 committed:
libmirage: refactor

guylamar2006 committed:
gcdemu: refactor

guylamar2006 committed:
cdemu: remove common-drv-attrs

keenanweaver commented on pkgs/by-name/cd/cdemu-client/package.nix:
```diff
@@ -1,17 +1,42 @@
 }:
+python3Packages.buildPythonApplication (finalAttrs: {
+  pname = "cdemu-client";
+  version = "3.2.5";
```
```suggestion
  version = "3.3.0";
```

keenanweaver commented on pkgs/by-name/cd/cdemu-client/package.nix:
```diff
@@ -1,17 +1,42 @@
-    url = "mirror://sourceforge/cdemu/${pname}-${version}.tar.xz";
-    inherit hash;
+    url = "mirror://sourceforge/cdemu/cdemu-client-${finalAttrs.version}.tar.xz";
+    hash = "sha256-py2F61v8vO0BCM18GCflAiD48deZjbMM6wqoCDZsOd8=";
```
```suggestion
    hash = "sha256-WiKm00ZKUDeHX6RII4+JKruAzUyWzZWyI0iZeDqksnA=";
```

keenanweaver commented on pkgs/by-name/cd/cdemu-daemon/package.nix:
```diff
@@ -0,0 +1,63 @@
+
+  src = fetchurl {
+    url = "mirror://sourceforge/cdemu/cdemu-daemon-${finalAttrs.version}.tar.xz";
+    hash = "sha256-EKh2G6RA9Yq46BpTAqN2s6TpLJb8gwDuEpGiwdGcelc=";
```
```suggestion
  version = "3.3.0";

  src = fetchurl {
    url = "mirror://sourceforge/cdemu/cdemu-daemon-${finalAttrs.version}.tar.xz";
    hash = "sha256-AYHjiOAQdu685gc6p0j2QNtCmTYTWix1kzWQZYvGPWU=";
```

keenanweaver commented on pkgs/by-name/gc/gcdemu/package.nix:
```diff
@@ -0,0 +1,69 @@
+
+  src = fetchurl {
+    url = "mirror://sourceforge/cdemu/gcdemu-${finalAttrs.version}.tar.xz";
+    hash = "sha256-w4vzKoSotL5Cjfr4Cu4YhNSWXJqS+n/vySrwvbhR1zA=";
```
```suggestion
  version = "3.3.0";

  src = fetchurl {
    url = "mirror://sourceforge/cdemu/gcdemu-${finalAttrs.version}.tar.xz";
    hash = "sha256-C9d4Rv47kQhs2kbTCwAUcdm+dcljA8IVkwhLJHJpUS0=";
```

keenanweaver commented on pkgs/by-name/im/image-analyzer/package.nix:
```diff
@@ -0,0 +1,71 @@
+
+  src = fetchurl {
+    url = "mirror://sourceforge/cdemu/image-analyzer-${finalAttrs.version}.tar.xz";
+    hash = "sha256-7I8RUgd+k3cEzskJGbziv1f0/eo5QQXn62wGh/Y5ozc=";
```
```suggestion
  version = "3.3.0";

  src = fetchurl {
    url = "mirror://sourceforge/cdemu/image-analyzer-${finalAttrs.version}.tar.xz";
    hash = "sha256-N2aufwYEBVx7z2Vo7Qi4DP2MsDXXr5LrQdeNYOtNGnU=";
```

keenanweaver commented on pkgs/by-name/li/libmirage/package.nix:
```diff
@@ -16,23 +15,18 @@
+
+  src = fetchurl {
+    url = "mirror://sourceforge/cdemu/libmirage-${finalAttrs.version}.tar.xz";
+    hash = "sha256-+T5Gu3VcprCkSJcq/kTySRnNI7nc+GbRtctLkzPhgK4=";
```
```suggestion
  version = "3.3.1";

  src = fetchurl {
    url = "mirror://sourceforge/cdemu/libmirage-${finalAttrs.version}.tar.xz";
    hash = "sha256-mstOGdmKJXRUrQA5F1DZGqVuY+f25Q5ZpdOXPx4MZRI=";
```

keenanweaver reviewed (changes requested):
Can you bump to the latest version of the components? cdemu is currently broken on the 7.0 kernel, and the updates probably resolve it.

Might want to look into a custom update script as well, so the next time it breaks, the bot should have a nice PR ready to merge for all the components.


For vhba:


``` nix
{
  lib,
  stdenv,
  fetchurl,
  kernel,
  kernelModuleMakeFlags,
}:

stdenv.mkDerivation (finalAttrs: {
  pname = "vhba";
  version = "20260313";

  src = fetchurl {
    url = "mirror://sourceforge/cdemu/vhba-module-${finalAttrs.version}.tar.xz";
    hash = "sha256-KTADv12dwrOG2w0F9ZXFVINVpTXW38Bv03n9mLsZAXQ=";
  };

  makeFlags = kernelModuleMakeFlags ++ [
    "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
    "INSTALL_MOD_PATH=$(out)"
  ];
  nativeBuildInputs = kernel.moduleBuildDependencies;

  meta = {
    description = "Provides a Virtual (SCSI) HBA";
    homepage = "https://cdemu.sourceforge.io/about/vhba/";
    platforms = lib.platforms.linux;
    license = lib.licenses.gpl2Plus;
    maintainers = with lib.maintainers; [ bendlas ];
  };
})

```

guylamar2006 committed:
cdemu-client: add updateScript

guylamar2006 committed:
cdemu-client: 3.2.5 -> 3.3.0

guylamar2006 committed:
cdemu-daemon: add updateScript

guylamar2006 committed:
cdemu-daemon: 3.2.7 -> 3.3.0

guylamar2006 committed:
gcdemu: add updateScript

guylamar2006 committed:
gcdemu: 3.2.6 -> 3.3.0

guylamar2006 committed:
libmirage: add updateScript

guylamar2006 committed:
libmirage: 3.2.10 -> 3.3.1

guylamar2006 committed:
image-analyzer: add updateScript

guylamar2006 committed:
image-analyzer: 3.2.6 -> 3.3.0

guylamar2006 committed:
linuxPackages.vhba: add updateScript

guylamar2006 committed:
linuxPackages.vhba: 20250329 -> 20260313

guylamar2006 committed:
linuxPackages.vhba: move kernel module to by-name

keenanweaver reviewed (approved):
Thanks! Looks good to me. Did not test on my machine, however. 

Regarding the update scripts, I can see some package drift happening the next go around, e.g. bot opens a PR for libmirage which gets merged, but a vhba PR happens a cycle or two later, which doesn't get merged for another few weeks after that. Is it possible to have one update script at say `cdemu-daemon` that touches multiple separate derivations? That'd prevent the issue from happening, but also can open another can of worms.

Do you like the changes, @bendlas?

## `nixpkgs-review` result for [#458540](https://github.com/NixOS/nixpkgs/pull/458540)

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 458540`
Commit: [`97a402f9191b8f7f154c9c328bb367bf7d682837`](https://github.com/NixOS/nixpkgs/commit/97a402f9191b8f7f154c9c328bb367bf7d682837) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/97a402f9191b8f7f154c9c328bb367bf7d682837..pull/458540/head))
Merge: [`b93485617a21d034e3ea6b17b65cd9033b390510`](https://github.com/NixOS/nixpkgs/commit/b93485617a21d034e3ea6b17b65cd9033b390510)

Logs: https://github.com/keenanweaver/nixpkgs-review-gha/actions/runs/24669855423


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 18 packages built:</summary>
  <ul>
    <li>cdemu-client</li>
    <li>cdemu-daemon</li>
    <li>gcdemu</li>
    <li>highscore</li>
    <li>highscore-unwrapped</li>
    <li>image-analyzer</li>
    <li>libmirage</li>
    <li>linuxKernel.packages.linux_5_10.vhba</li>
    <li>linuxKernel.packages.linux_5_15.vhba</li>
    <li>linuxKernel.packages.linux_6_1.vhba</li>
    <li>linuxKernel.packages.linux_6_12.vhba</li>
    <li>linuxPackages.vhba (linuxKernel.packages.linux_6_18.vhba)</li>
    <li>linuxKernel.packages.linux_6_19.vhba</li>
    <li>linuxKernel.packages.linux_6_6.vhba</li>
    <li>linuxPackages_latest.vhba (linuxKernel.packages.linux_7_0.vhba)</li>
    <li>linuxPackages_xanmod.vhba (linuxKernel.packages.linux_xanmod.vhba)</li>
    <li>linuxPackages_xanmod_latest.vhba (linuxKernel.packages.linux_xanmod_latest.vhba, linuxPackages_xanmod_stable.vhba)</li>
    <li>linuxPackages_zen.vhba (linuxKernel.packages.linux_zen.vhba)</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 15 packages built:</summary>
  <ul>
    <li>cdemu-client</li>
    <li>cdemu-daemon</li>
    <li>gcdemu</li>
    <li>highscore</li>
    <li>highscore-unwrapped</li>
    <li>image-analyzer</li>
    <li>libmirage</li>
    <li>linuxKernel.packages.linux_5_10.vhba</li>
    <li>linuxKernel.packages.linux_5_15.vhba</li>
    <li>linuxKernel.packages.linux_6_1.vhba</li>
    <li>linuxKernel.packages.linux_6_12.vhba</li>
    <li>linuxPackages.vhba (linuxKernel.packages.linux_6_18.vhba)</li>
    <li>linuxKernel.packages.linux_6_19.vhba</li>
    <li>linuxKernel.packages.linux_6_6.vhba</li>
    <li>linuxPackages_latest.vhba (linuxKernel.packages.linux_7_0.vhba)</li>
  </ul>
</details>

bendlas commented:
> Regarding the update scripts, I can see some package drift happening the next go around, e.g. bot opens a PR for libmirage which gets merged, but a vhba PR happens a cycle or two later, which doesn't get merged for another few weeks after that. Is it possible to have one update script at say `cdemu-daemon` that touches multiple separate derivations?

Not sure if it's possible, just to add some context: cdemu packages are not always released together and don't always have the same version numbers. Some apparent intconsistencies may be due to that ...

> Do you like the changes, @bendlas?

:heavy_check_mark:

alyssais commented on pkgs/applications/emulators/cdemu/analyzer.nix:
```diff
@@ -1,54 +0,0 @@
-{
```
Could you please rename and refactor in separate commits, so git knows about the renames?

guylamar2006 committed:
cdemu: inline common-drv-attrs and refactor

guylamar2006 committed:
cdemu-*: migrate to by-name

guylamar2006 committed:
cdemu-*: add updateScript

guylamar2006 committed:
cdemu-*: update to latest versions

cdemu-daemon: 3.2.7 -> 3.3.0

cdemu-client: 3.2.5 -> 3.3.0

gcdemu: 3.2.6 -> 3.3.0

image-analyzer: 3.2.6 -> 3.3.0

libmirage: 3.2.10 -> 3.3.1

guylamar2006 committed:
linuxPackages.vhba: add updateScript

guylamar2006 committed:
linuxPackages.vhba: migrate kernel module to by-name

keenanweaver reviewed (approved):
Thanks! Should be good to merge.

## `nixpkgs-review` result for [#458540](https://github.com/NixOS/nixpkgs/pull/458540)

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha) ([`04076f7`](https://github.com/Defelo/nixpkgs-review-gha/commit/04076f7df04b66831d730e9cb4f24e1664b6cb76))

Command: `nixpkgs-review pr 458540`
Commit: [`b3bd938d3d5b014efc99844826cb779babdf8d9d`](https://github.com/NixOS/nixpkgs/commit/b3bd938d3d5b014efc99844826cb779babdf8d9d) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/b3bd938d3d5b014efc99844826cb779babdf8d9d..pull/458540/head))
Merge: [`73e90ba82b0bc55e4576b7e39550e6f3b4f38d10`](https://github.com/NixOS/nixpkgs/commit/73e90ba82b0bc55e4576b7e39550e6f3b4f38d10)

Logs: https://github.com/keenanweaver/nixpkgs-review-gha/actions/runs/27215261826


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>cdemu-client</li>
    <li>cdemu-daemon</li>
    <li>gcdemu</li>
    <li>highscore</li>
    <li>highscore-unwrapped</li>
    <li>image-analyzer</li>
    <li>libmirage</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>cdemu-client</li>
    <li>cdemu-daemon</li>
    <li>gcdemu</li>
    <li>highscore</li>
    <li>highscore-unwrapped</li>
    <li>image-analyzer</li>
    <li>libmirage</li>
  </ul>
</details>
